src/main/java/Deadline.java

```diff
@@ -1,52 +1,61 @@
 public class Deadline extends Task implements FormatChecker {
     private static final String DEADLINE_MARK = "[D]";
-    private String by;
+    private static final String taskType = "D";
+    private String time;
 
     /**
      * Default constructor for Deadline instance
      *
      * @param description description of Deadline
-     * @param by time of deadline
+     * @param time        time of deadline
      */
-    public Deadline(String description, String by) {
+    public Deadline(String description, String time) {
         super(description);
-        this.by = by;
+        this.time = time;
     }
 
     /**
      * Constructor for a raw string input of deadline
      *
      * @param complexString raw string containing description and deadline
      */
-    public Deadline(String complexString) throws WrongCommandFormatException{
+    public Deadline(String complexString) throws WrongCommandFormatException {
         super();
         try {
             int identifierIndex = FormatChecker.findIdentifierIndex(DEADLINE_IDENTIFIER, complexString);
             this.description = complexString.substring(0, identifierIndex);
             FormatChecker.checkNullString(this.description);
-            this.by = complexString.substring(identifierIndex + TIME_IDENTIFIER_LENGTH);
-            FormatChecker.checkNullString(this.by);
+            this.time = complexString.substring(identifierIndex + TIME_IDENTIFIER_LENGTH);
+            FormatChecker.checkNullString(this.time);
             this.isDone = false;
         } catch (WrongCommandFormatException e) {
             throw new WrongCommandFormatException();
         }
     }
 
-    public String getBy() {
-        return this.by;
-    }
-
-    public void setBy(String by) {
-        this.by = by;
+    public Deadline(String description, Boolean isDone, String time) {
+        this.description = description;
+        this.isDone = isDone;
+        this.time = time;
     }
 
     @Override
     public String getStatusIcon() {
         return DEADLINE_MARK + super.getStatusIcon();
     }
 
+    @Override
+    public String getTaskType() {
+        return taskType;
+    }
+
+    @Override
+    public String getTime() {
+        return this.time;
```

Comment:
Perhaps you can avoid the unnecessary use of `this`.

```suggestion
        return time;
```
https://se-education.org/guides/conventions/java/index.html#variables